CORE-1139 | Support grpc headsampling (#5001)
odigos-io/odigos: docs/snippets/shared/pipeline/sampling/examples.mdx

</Note>

## 6. Drop noisy gRPC health checks

**Goal:** Drop standard gRPC liveness/readiness checks (`grpc.health.v1.Health.Check`, `Watch`, ...) on the `inventory` Deployment. This is the gRPC counterpart of [Example 1](#1-drop-a-noisy-custom-endpoint), and shows how setting `service` alone scopes a rule to every method of a whole gRPC service.

```yaml drop-grpc-health-checks.yaml
apiVersion: odigos.io/v1alpha1
kind: Sampling
metadata:
  name: drop-grpc-health-checks
  namespace: odigos-system
spec:
  name: Drop gRPC health checks
  noisyOperations:
    - name: Drop gRPC health checks
      notes: Standard liveness/readiness pings; not useful for tracing.
      percentageAtMost: 0
      sourceScopes:
        - workloadNamespace: odimall
          workloadKind: Deployment
          workloadName: inventory
      operation:
        grpcServer:
          service: grpc.health.v1.Health
```

<Tip>

>   Because `service` and `method` are independent, a single `service:` line filters every method on a service. Add `method: ListItems` to scope to one specific RPC instead, or combine with `grpcClient.serverAddress` to drop outgoing calls to a particular backend.

Reviewer: ```suggestion
  Because `service` and `method` are independent, configuring `service` without `method` will match every method on a service. Add `method: <method_name>` to scope to one specific RPC instead, or combine with `grpcClient.serverAddress` to drop outgoing calls to a particular backend.
```
